### PR TITLE
Graffiti ordered index

### LINKED ISSUE #8181: Graffiti - Ordered list option for yaml file.
I would like to request an option to read a list of items from a graffiti file. Currently there is only an option for a default and a random list, but I would like a list that goes in order. Example below:

```
ordered:
  - graffitiwall:300:500:#000000
  - graffitiwall:300:50:1#000000
```

### PR DESCRIPTION
**Pull request by qix**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

This is based on top of #8189 by @pinglamb. I extended it so that instead of resetting upon validator restart the index is stored in the database. Alongside it I store a hash of the graffiti file so that the index can be reset whenever the file is updated.

**Which issues(s) does this PR fix?**

Fixes #8181 and fixes #8189

**Other notes for review**

I left the behavior of cycling through the graffiti untouched, however I think it would be better if each was only emitted once. That way `ordered` could be combined with `random`/`default` which would take effect once all the ordered entries have been emitted.

### CODE REVIEW

**Comment by rkapka** [comment]
@qix  `bazel run //:gazelle -- fix` is needed to fix dependencies. Otherwise the build will keep failing.

**Comment by qix** [comment]
> @qix `bazel run //:gazelle -- fix` is needed to fix dependencies. Otherwise the build will keep failing.
> 
> And of course thank you for the contribution, my friend!

Thanks! Might be worth mentioning that in the `CONTRIBUTING` doc. Just ran it and the only thing I missed was ordering in one of the BUILD fles

File: validator/db/kv/graffiti.go
Comment (rauljordan) [inline validator/db/kv/graffiti.go:13]: Small thing not necessary to fix, but can be simplified to `return s.db.Update`

**Review by rauljordan — DISMISSED** [review]
Nice job and unit tests!

**Review by terencechain — CHANGES_REQUESTED** [review]
Looks like @prestonvanloon has concerns about saving graffiti index to DB.

Let's have him take one last look before merging

https://github.com/prysmaticlabs/prysm/pull/8189#discussion_r562089786

**Comment by rauljordan** [comment]
Please take a look at conflicts @qix. Once resolved, you will also need to run `bazel run //:gazelle -- fix`

**Comment by qix** [comment]
Rebased

The concerns @prestonvanloon raised were about how the index might be reset which I believe this solves (by tying it directly to the hash of the graffiti source). I'll wait for the okay from him though.

**Review by prestonvanloon — APPROVED** [review]
The design of this looks good. I'm happy with the solution. Raul had a suggestion, but I approve

**Comment by qix** [comment]
Thanks @prestonvanloon.

Sorry I'm not entirely used to this workflow, I included the fix Raul gave and rebased on top of master but that dismissed your approval. It looks like I'll need another stamp from @terencechain as well since they requested changes.

**Review by terencechain — APPROVED** [review]

**Comment by qix** [comment]
Awesome! Thanks a lot for the pleasant review y'all :)

Hopefully you'll see a bit more from me in the future.

**Pull request merged**
